src/OrchardCore.Modules/OrchardCore.Media.Azure/Services/MediaBlobContainerTenantEvents.cs

```diff
@@ -56,28 +56,65 @@ public override async Task ActivatingAsync()
+        // Only carry out removal if options are valid.
+        if (!_options.IsConfigured() || (!_options.RemoveFilesFromBasePath && !_options.RemoveContainer))
         {
             return;
         }
 
-        try
+        var blobContainer = new BlobContainerClient(_options.ConnectionString, _options.ContainerName);
+        if (_options.RemoveContainer)
         {
-            var blobContainer = new BlobContainerClient(_options.ConnectionString, _options.ContainerName);
+            try
+            {
+                var response = await blobContainer.DeleteIfExistsAsync();
+                if (!response.Value)
+                {
+                    _logger.LogError("Unable to remove the Azure Media Storage Container {ContainerName}.", _options.ContainerName);
 
-            var response = await blobContainer.DeleteIfExistsAsync();
-            if (!response.Value)
+                    context.ErrorMessage = S["Unable to remove the Azure Media Storage Container '{0}'.", _options.ContainerName];
+                }
+            }
+            catch (RequestFailedException ex)
             {
-                _logger.LogError("Unable to remove the Azure Media Storage Container {ContainerName}.", _options.ContainerName);
-                context.ErrorMessage = S["Unable to remove the Azure Media Storage Container '{0}'.", _options.ContainerName];
+                _logger.LogError(ex, "Failed to remove the Azure Media Storage Container {ContainerName}.", _options.ContainerName);
+
+                context.ErrorMessage = S["Failed to remove the Azure Media Storage Container '{0}'.", _options.ContainerName];
+                context.Error = ex;
+            }
+
+            // If both deletion options are set, return here to avoid errors when trying to delete files from a non-existent container.
+            if (_options.RemoveFilesFromBasePath)
+            {
+                return;
             }
         }
-        catch (RequestFailedException ex)
+
+        if (_options.RemoveFilesFromBasePath)
         {
-            _logger.LogError(ex, "Failed to remove the Azure Media Storage Container {ContainerName}.", _options.ContainerName);
-            context.ErrorMessage = S["Failed to remove the Azure Media Storage Container '{0}'.", _options.ContainerName];
-            context.Error = ex;
+            try
+            {
+                await foreach (var blobItem in blobContainer.GetBlobsAsync(prefix: _options.BasePath))
+                {
+                    var response = await blobContainer.DeleteBlobIfExistsAsync(blobItem.Name);
+                    if (!response.Value)
+                    {
+                        _logger.LogError("File removal process failed on file {ItemName}.", blobItem.Name);
+
+                        context.ErrorMessage = S["File removal process failed on file {ItemName}.", blobItem.Name];
+
+                        // Also stop the removal process if a file fails.
+                        break;
```

Comment:
Don't we want to continue here? This might be a transient error only. But concatenate the error message if multiple files are failing.

Reply:
Is it better practice to keep going in such cases?

Reply:
I guess the whole process stops if there is an error. So you're correct with stopping here too.